### PR TITLE
Add DisableSourceLinkUrlTranslation property

### PR DESCRIPTION
**Pull request by MattGal**
See https://github.com/dotnet/arcade/issues/5446 for context

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/RepositoryInfo.targets [hunk 1]
```diff
@@ -43,11 +44,14 @@
...
       git@ssh.dev.azure.com:v3/dnceng/internal/dotnet-arcade-services
     -->
+    <!-- Set DisableSourceLinkUrlTranslation to true when building a tool for internal use where sources only come from internal URIs -->
+    <DisableSourceLinkUrlTranslation Condition="'$(DisableSourceLinkUrlTranslation)' == ''">false</DisableSourceLinkUrlTranslation>
     <_TranslateUrlPattern>(https://dnceng%40dev\.azure\.com/dnceng/internal/_git|https://dev\.azure\.com/dnceng/internal/_git|https://dnceng\.visualstudio\.com/internal/_git|dnceng%40vs-ssh\.visualstudio\.com:v3/dnceng/internal|git%40ssh\.dev\.azure\.com:v3/dnceng/internal)/([^/-]+)-(.+)</_TranslateUrlPattern>
     <_TranslateUrlReplacement>https://github.com/$2/$3</_TranslateUrlReplacement>
   </PropertyGroup>
 
   <Target Name="_TranslateAzureDevOpsUrlToGitHubUrl"
+          Condition=" '$(DisableSourceLinkUrlTranslation)' == 'false' "
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/RepositoryInfo.targets:54]: Style: No spaces around the expression.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/RepositoryInfo.targets [hunk 2]
```diff
@@ -19,7 +19,8 @@
   </ItemGroup>
 
   <PropertyGroup Condition="'$(EnableSourceControlManagerQueries)' != 'true' and '$(RepositoryUrl)' == ''">
-    <RepositoryUrl Condition="'$(BUILD_REPOSITORY_URI)' != ''">$([System.Text.RegularExpressions.Regex]::Replace($(BUILD_REPOSITORY_URI), $(_TranslateUrlPattern), $(_TranslateUrlReplacement)))</RepositoryUrl>
+    <RepositoryUrl Condition="'$(BUILD_REPOSITORY_URI)' != '' And '$(DisableSourceLinkUrlTranslation)' != 'true' ">$([System.Text.RegularExpressions.Regex]::Replace($(BUILD_REPOSITORY_URI), $(_TranslateUrlPattern), $(_TranslateUrlReplacement)))</RepositoryUrl>
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/RepositoryInfo.targets:22]: Style: lower-case `and`

**Comment by tmat** [comment]
I wonder if we could replace this translation by using git's `insteadOf` configuration option. If this option is set by the pipeline when the repo is cloned SourceLink will respect it and perform the prefix replacement.

**Comment by MattGal** [comment]
> I wonder if we could replace this translation by using git's `insteadOf` configuration option. If this option is set by the pipeline when the repo is cloned SourceLink will respect it and perform the prefix replacement.

Something to consider for later, sure.

**Pull request merged**